adam-core/src/main/scala/org/bdgenomics/adam/rdd/features/FeatureRDD.scala

```diff
@@ -173,44 +238,35 @@ class FeatureRDDFunctions(featureRDD: RDD[Feature]) extends Serializable with Lo
     genes
   }
 
-  def reassignParentIds(): RDD[Feature] = {
-    def reassignParentId(f: Feature): Feature = {
-      val fb = Feature.newBuilder(f)
-
-      val featureId: Option[String] = Option(f.getFeatureId)
-      val geneId: Option[String] = Option(f.getGeneId)
-      val transcriptId: Option[String] = Option(f.getTranscriptId)
-      val exonNumber: Option[String] = Option(f.getAttributes.get("exon_number"))
-      val exonId: Option[String] = Option(f.getExonId) match {
-        case Some(e) => Option(e)
-        case None    => transcriptId.flatMap(t => exonNumber.map(e => t + "_" + e))
-      }
-
-      val (featureIdOpt, parentIdOpt) =
-        f.getFeatureType match {
-          case "gene"        => (geneId, None)
-          case "transcript"  => (transcriptId, geneId)
-          case "exon"        => (exonId, transcriptId)
-          case "CDS" | "UTR" => (featureId, transcriptId)
-          case _             => (featureId, None)
-        }
-      // replace featureId and parentId if defined
-      featureIdOpt.foreach(fb.setFeatureId)
-      parentIdOpt.foreach(parentId => fb.setParentIds(List[String](parentId)))
+  /**
+   * @return Returns this RDD, where the Gene structure was rebuilt.
+   */
+  def toGenes(): GeneRDD = {
+    GeneRDD(rddToGeneRDD, sequences)
+  }
 
-      fb.build()
-    }
-    featureRDD.map(reassignParentId)
+  /**
+   * @param newRdd The RDD to replace the underlying RDD with.
+   * @return Returns a new FeatureRDD with the underlying RDD replaced.
+   */
+  protected def replaceRdd(newRdd: RDD[Feature]): FeatureRDD = {
+    copy(rdd = newRdd)
   }
 
-  def filterByOverlappingRegion(query: ReferenceRegion): RDD[Feature] = {
-    def overlapsQuery(rec: Feature): Boolean =
-      rec.getContigName == query.referenceName &&
-        rec.getStart < query.end &&
-        rec.getEnd > query.start
-    featureRDD.filter(overlapsQuery)
+  /**
+   * @param elem The Feature to get an underlying region for.
+   * @return Since a feature maps directly to a single genomic region, this
+   *   method will always return a Seq of exactly one ReferenceRegion.
+   */
+  protected def getReferenceRegions(elem: Feature): Seq[ReferenceRegion] = {
+    Seq(ReferenceRegion(elem))
   }
 
+  /**
+   * Saves this FeatureRDD to a GTF file.
```

Comment:
How about
`Save this FeatureRDD in GTF format.`
and
`@param fileName The path to save GTF formatted text file(s) to.`

and similar for below?